retroshare-gui/src/gui/Posted/PhotoView.cpp

```diff
@@ -0,0 +1,106 @@
+ * published by the Free Software Foundation, either version 3 of the          *
+ * License, or (at your option) any later version.                             *
+ *                                                                             *
+ * This program is distributed in the hope that it will be useful,             *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                *
+ * GNU Affero General Public License for more details.                         *
+ *                                                                             *
+ * You should have received a copy of the GNU Affero General Public License    *
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.       *
+ *                                                                             *
+ *******************************************************************************/
+
+#include "PhotoView.h"
+
+#include <QMenu>
+#include <QFileDialog>
+#include <QMessageBox>
+
+#include "gui/gxs/GxsIdDetails.h"
+#include "gui/RetroShareLink.h"
+
+#include <retroshare/rsidentity.h>
+#include <retroshare/rsposted.h>
+
+/** Constructor */
+PhotoView::PhotoView(QWidget *parent)
+: QDialog(parent),
+	ui(new Ui::PhotoView)
+{
+  /* Invoke the Qt Designer generated object setup routine */
+  ui->setupUi(this);
+
+  setAttribute(Qt::WA_DeleteOnClose, true);
+  
+  connect(ui->shareButton, SIGNAL(clicked()), this, SLOT(copyMessageLink()));
+}
+
+/** Destructor */
+PhotoView::~PhotoView()
+{
+	delete ui;
+}
+
+void PhotoView::setPixmap(const QPixmap& pixmap) 
+{
+	QPixmap sqpixmap = pixmap.scaled(640,480, Qt::KeepAspectRatio, Qt::SmoothTransformation);
+
+	if(pixmap.width() > 800 || pixmap.height() > 700)
+		ui->photoLabel->setPixmap(sqpixmap);
+	else
+		ui->photoLabel->setPixmap(pixmap);
+}
+
+void PhotoView::setTitle(const QString& text) 
+{
+	ui->titleLabel->setText(text);
+}
+
+void PhotoView::setName(const RsGxsId& authorID) 
+{
+	ui->nameLabel->setId(authorID);
+	
+	RsIdentityDetails idDetails ;
+	rsIdentity->getIdDetails(authorID,idDetails);
+
+	QPixmap pixmap ;
+
+	if(idDetails.mAvatar.mSize == 0 || !GxsIdDetails::loadPixmapFromData(idDetails.mAvatar.mData, idDetails.mAvatar.mSize, pixmap,GxsIdDetails::SMALL))
+			pixmap = GxsIdDetails::makeDefaultIcon(authorID,GxsIdDetails::SMALL);
+			
+	ui->avatarWidget->setPixmap(pixmap);
+}
+
+void PhotoView::setTime(const QString& text) 
+{
+	ui->timeLabel->setText(text);
+}
+
+void PhotoView::setGroupId(const RsGxsGroupId &groupId) 
+{
+	mGroupId = groupId;
+}
+
+void PhotoView::setMessageId(const RsGxsMessageId& messageId) 
+{
+	mMessageId = messageId ;
+}
+
+void PhotoView::copyMessageLink()
+{
+	RetroShareLink link = RetroShareLink::createGxsMessageLink(RetroShareLink::TYPE_POSTED, mGroupId, mMessageId, ui->titleLabel->text());
+
+	if (link.valid()) {
```

Comment:
The user needs some indication that something happened, a messagebox.

Reply:
Yes i had that in my mind bit thinked maybe o disable that button 
rs links for posted messages still not works yet